mta/entities/rules_configuration.py

```diff
@@ -0,0 +1,181 @@
+from widgetastic_patternfly4 import Button
+from widgetastic_patternfly4 import Dropdown
+from widgetastic_patternfly4 import Pagination
+from widgetastic_patternfly4.table import Table
+
+from mta.base.application.implementations.web_ui import MTANavigateStep
+from mta.base.application.implementations.web_ui import ViaWebUI
+from mta.base.modeling import BaseCollection
+from mta.entities import BaseLoggedInPage
+from mta.utils import conf
+from mta.utils.ftp import FTPClientWrapper
+from mta.widgetastic import DropdownMenu
+from mta.widgetastic import HiddenFileInput
+from mta.widgetastic import Input
+from mta.widgetastic import MTACheckboxSelect
+from mta.widgetastic import MTATab
+
+
+class SystemRulesView(BaseLoggedInPage):
+    """This view represents System rules tab page"""
+
+    show_all_rules = Checkbox(".//input[@id='showAllRules']")
+    table = Table(
+        locator='.//table[contains(@aria-label, "Table")]',
+        column_widgets={
+            "Provider ID": Button(locator='.//th[@data-label="Provider ID"]/button'),
+            "Number of rules": Button(locator=".//th[@data-label='Number of rules']/button"),
+            4: Dropdown(),
+        },
+    )
+    filter_type_selector = DropdownMenu(locator='.//div[@class="pf-c-toolbar__item"]')
+    filter_by = MTACheckboxSelect(
+        locator='.//span[@class="pf-c-select__toggle-arrow"]//parent::button['
+        'contains(@aria-labelledby, "Filter by Source")]/parent::div")'
+    )
+
+    clear = Text('.//button[contains(text(), "Clear all filters")]')
+
+    def clear_filters(self):
+        if self.clear.is_displayed:
+            self.clear.click()
+
+    def search(self, search_value, filter_type="Source", clear_filters=False):
+        """Fill input box with 'search_value', use 'filter_type' to choose filter selector.
+        If no filter_type is entered then the default for page is used.
+        """
+        if clear_filters:
+            self.clear_filters()
+        if filter_type:
+            self.filter_type_selector.item_select(filter_type)
+        self.filter_by.fill(search_value)
+
+
+class DeleteCustomRuleView(View):
+    title = Text(locator=".//h1[contains(normalize-space(.), 'Delete')]")
+    fill_strategy = WaitFillViewStrategy("35s")
+
+    delete_button = Button("Delete")
+    cancel_button = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return self.delete_button.is_displayed
+
+
+class CustomRulesView(BaseLoggedInPage):
+    """This view represents Custom rules tab page"""
+
+    add_rule_button = Button("Add rule")
+    search = Input(locator=".//input[@aria-label='Filter by short path']")
+    table = Table(
+        locator='.//table[contains(@aria-label, "Table")]',
+        column_widgets={
+            "Short path": Button(locator='.//th[@data-label="Short path"]/button'),
+            4: Dropdown(),
+        },
+    )
+    delete_rules = View.nested(DeleteCustomRuleView)
+
+    @property
+    def is_displayed(self):
```

Comment:
I think we need to check for more than one thing here because in jenkins run buttons render faster and page renders later and test fails . So also add check for "table displayed"